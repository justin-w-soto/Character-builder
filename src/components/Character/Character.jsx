import React from 'react'

export const Character = ({head, middle, pants}) => {
    return (
        <section style={{display: 'flex', flexDirection: 'column'}}>
            <article 
            className='head'
            style={{
                backgroundImage: `url(./${head}-head.jpg)`,
                width: 200,
                height: 200,
                backgroundSize: 'cover',
                backgroundRepeat: 'no-repeat',
            }}
            ></article>

            <article 
            className='middle'
            style={{
                backgroundImage: `url(./${middle}-middle.jpg)`,
                width: 200,
                height: 200,
                backgroundSize: 'cover',
                backgroundRepeat: 'no-repeat',
            }}
            ></article>

            <article 
            className='pants'
            style={{
                backgroundImage: `url(./${pants}-pants.jpg)`,
                width: 200,
                height: 200,
                backgroundSize: 'cover',
                backgroundRepeat: 'no-repeat',
            }}
            ></article>
        </section>
    )
}
