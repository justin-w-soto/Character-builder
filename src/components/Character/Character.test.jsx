import { render, screen } from '@testing-library/react';
import Character from './Character';;

it('should display the Character component', () => {
    const {container} = render(<Character head='alien' middle='cool' pants='bread' />);
  
    const head = screen.getByLabelText(/Head/);
    const middle = screen.getByLabelText(/Middle/);
    const pants = screen.getByLabelText(/Pants/);

    expect(head).toHaveStyle(`background-image: url(./alien-head.jpg)`);
    expect(middle).toHaveStyle(`background-image: url(./cool-middle.jpg)`);
    expect(pants).toHaveStyle(`background-image: url(./bread-pants.jpg)`)

  
    expect(container).toMatchSnapshot();
  });