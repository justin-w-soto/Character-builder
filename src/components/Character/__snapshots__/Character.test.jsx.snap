// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display the Character component 1`] = `
<div>
  <section
    style="display: flex; flex-direction: column;"
  >
    <article
      aria-label="Head"
      class="head"
      style="background-image: url(./alien-head.jpg); border: 2px solid brown; border-radius: 2rem; height: 200px; background-size: cover; background-repeat: no-repeat; margin: 2rem;"
    >
      Head
    </article>
    <article
      aria-label="Middle"
      class="middle"
      style="background-image: url(./cool-middle.jpg); border: 2px solid brown; border-radius: 2rem; height: 200px; background-size: cover; background-repeat: no-repeat; margin: 2rem;"
    >
      middle
    </article>
    <article
      aria-label="Pants"
      class="pants"
      style="background-image: url(./bread-pants.jpg); border: 2px solid brown; border-radius: 2rem; height: 200px; background-size: cover; background-repeat: no-repeat; margin: 2rem;"
    >
      pants
    </article>
  </section>
</div>
`;
